### DISCUSSION TITLE
[TUTORIAL] How To Troubleshoot "Confirm you are on the latest version using  yt-dlp -U" Problem on 1.90 SEAL and under.

### DISCUSSION THREAD

**Discussion by nyatnyat** [JunkFood02/Seal#673] [General]
I found this troubleshooting by accident when i was having frustration, that i'm unable to download my video as usual, and having that nasty yt-dlp-U. Little did i know the troubleshoot were already there in first place lol. 

Here's tutorial on fix said yt-dlp-U bugger, hope this is working for you. Feel free to correct me.

1. Go to Seal Settings and choose "Custom Command"
![Screenshot_20230313_135831](https://user-images.githubusercontent.com/101979888/224622722-56cf7db4-7d37-4499-b28e-95dbd4cbdc8b.jpg)

2. Enabled the custom command of yt-dlp-U by selecting the "Choose Custom Command"
![Screenshot_20230313_135835 (1)](https://user-images.githubusercontent.com/101979888/224623129-b5b06013-aef7-4c7c-8d90-ac60fa7d387d.jpg)

3. Prepare the video that you wanted to download using Seal 
![gambar_2023-03-13_141902342](https://user-images.githubusercontent.com/101979888/224622389-f7fd97ca-5cc4-45cb-babc-85e38190e99d.png)

4. Using Youtube Share button, choose Seal on your selectable sharing platform or whatever lol
![Screenshot_20230313_135951](https://user-images.githubusercontent.com/101979888/224623420-1fc814de-7431-49f2-a256-8e17b6b612a4.jpg)

5. There might be this landing page if you've enabled it on Settings, choose "Custom Command" on Advanced section
![Screenshot_20230313_135956](https://user-images.githubusercontent.com/101979888/224623669-e20733ff-71a2-49bc-adaa-51964f91c804.jpg)

6. And Done! You could patiently wait and see the progress of your downloaded video on Gallery later.
![Screenshot_20230313_140012](https://user-images.githubusercontent.com/101979888/224623876-671b218d-37d0-4230-bac5-fd5e9c4b92d6.jpg)

**Comment by JunkFood02** [comment]
You can simply click/long-press on the "yt-dlp version" in the general settings to update yt-dlp/configure the update preference of it

**Reply by nyatnyat** [reply]
Oh its Junk himself lol. And yeah for some reason my Seal still stuck on January edition lol. Nonetheless when i updated it to March version im still having that yt-dlp-U problem.

**Reply by JunkFood02** [reply]
the `yt-dlp -U` is merely a suggestion to update the yt-dlp to the latest version, and it appears in the error report no matter whether you're using the latest one.
Also, the auto-update for yt-dlp is enabled by default in v1.9.0. the app will check for updates on every launch. Thus, the errors are unlikely due to an outdated yt-dlp version now

**Reply by nyatnyat** [reply]
> the `yt-dlp -U` is merely a suggestion to update the yt-dlp to the latest version, and it appears in the error report no matter whether you're using the latest one.

Ahh now i understood, so should i change this tutorial to just merely update the yt-dlp-U or made new one lol? But Junk you should checkout bug where people still having this kind of problem eventhought they updated their yt-dlp -U

**Reply by JunkFood02** [reply]
It depends:
* `-U` means Update for sure. If there's something not working in Seal, since Seal is a GUI for yt-dlp, the same issue should be addressed in yt-dlp as well. The only solution is to wait for the upstream to release a new version for fixes.
* If the problem cannot be reproduced on differenent devices/internet environment etc., it seems like that we can't do anything about it either (e.g instagram/facebook issues)
* If the problem can be only reproduced with Seal but not with yt-dlp running on a PC, it's probably a bug in Seal's codebase, file a bug report for it
